loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,384 @@
+/*
+ * ===========================================================================
+ * Loom SDK
+ * Copyright 2011, 2012, 2013
+ * The Game Engine Company, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================
+ */
+
+#include "platform/CCFileUtils.h"
+#include "loom/script/loomscript.h"
+#include "loom/common/core/log.h"
+#include "loom/common/utils/utByteArray.h"
+#include "loom/script/native/lsNativeDelegate.h"
+#include "loom/vendor/sqlite3/sqlite3.h"
+#include "loom/common/utils/utByteArray.h"
+
+lmDefineLogGroup(gSQLiteGroup, "loom.sqlite", 1, LoomLogInfo);
+
+using namespace LS;
+
+
+//forward declaration of Statement
+class Statement;
+
+
+//SQLite Connection binding for Loomscript
+class Connection
+{
+protected:
+    char szDBName[128];
+    char szDBFullPath[1024];
+    sqlite3 *dbHandle;
+
+public:
+    static Connection *open(const char *database, int flags);
+    static const char *getVersion();
+
+    Statement *prepare(const char *query);
+    const char* getDBName() { return szDBName; }
+
+
+    int getErrorCode()
+    {
+        return sqlite3_errcode(dbHandle);
+    }
+  
+    const char* getErrorMessage()
+    {
+        return sqlite3_errmsg(dbHandle);
+    }
+      
+    int getlastInsertRowId()
+    {
+        //"NOTE: In SQLite the row ID is a 64-bit integer but for all practical 
+        //database sizes you can cast the 64 bit value to a 32-bit integer."
+        //
+        // - Some Guy on The Internet
+        //
+        int rowid = (int)sqlite3_last_insert_rowid(dbHandle);
+        return rowid;
+    }
+
+    void close()
```

Comment:
Not much choice unfortunately as Loomscript doesn't support 64bit ints :/
Will note the limitation interested LS function comment.
On 12 Feb 2015 5:32 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r24556165>:
>
> > +    {
> > +        return sqlite3_errmsg(dbHandle);
> > +    }
> > +
> > +    int getlastInsertRowId()
> > +    {
> > +        //"NOTE: In SQLite the row ID is a 64-bit integer but for all practical
> > +        //database sizes you can cast the 64 bit value to a 32-bit integer."
> > +        //
> > +        // - Some Guy on The Internet
> > +        //
> > +        int rowid = (int)sqlite3_last_insert_rowid(dbHandle);
> > +        return rowid;
> > +    }
> > +
> > +    void close()
>
> Yeah if it can error note it, even though it's a bad case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r24556165>.
>